### PR TITLE
set $NINJAFLAGS to make sure Ninja doesn't use all visible cores when building Qt5

### LINKED ISSUE #2076: Qt  Make/Ninja interaction
When building Qt5 5.12.8 I observed that the QtWebEngine part was very slow. It uses Ninja inside Make and for some reason Ninja only spawned one g++ process at a time. Now I did use something non-standard, namely using a Gentoo Prefix Ninja, which is a fork with support for the GNU Make jobserver (https://github.com/Kitware/ninja), and that was what caused this interaction.

However when putting ninja 1.10.0 as a builddependency I see this:
```
Checking for system ninja... no
```
and it then proceeds to build and use an included Ninja in (`qtwebengine/src/3rdparty/ninja/ninja`), which is Ninja version 1.8.2 in this case.
From a pure EB perspective I see two issues here:

1. The Qt make procedure has this check (also in 5.14.1): 
```     
        contains(version, "1.[7-9].*"): return(true)
        qtLog("Ninja version too old")
```
Ninja 1.10 fails this test and that's why it uses the included Ninja instead of the builddep.

2. Ninja by default (ie. not the one with GNU Make jobserver support) will gobble up all cores. If you want to avoid that you need to pass it an explicit `-jN` flag, via `prebuildopts = "NINJAFLAGS='-j8'"` (see https://patchwork.ozlabs.org/project/buildroot/patch/20180629143521.9018-1-gael.portay@savoirfairelinux.com/ )

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

fixes #2076

### CODE REVIEW

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.14.2-GCCcore-10.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u17a.bear.cluster - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/791b4cd9650eef20d139873cd0d20ebe for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.13.1-GCCcore-8.3.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u18a.bear.cluster - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/a14974e2a68ecbf2bdd7f6c30e3f2763 for a full test report.

**Comment by branfosj** [comment]
Test report by @branfosj
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.14.1-GCCcore-9.3.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0211u17b.bear.cluster - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6248 CPU @ 2.50GHz (cascadelake), Python 3.6.8
See https://gist.github.com/aed794b5ce7052e8161a4109a8488435 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.14.2-GCCcore-10.2.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3551.doduo.os - Linux RHEL 8.2, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/c01e96c444f316466995bef2b6d64270 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.10.1-foss-2018b.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node3117.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/9b2ca0234611b964310506efe567438a for a full test report.

**Comment by boegel** [comment]
@branfosj suggested that we should find a more general place to add `-j` to `$NINJAFLAGS`, perhaps even in EasyBuild framework, since it likely also affects other software.

Just doing so in the `MesonNinja` generic easyblock is not good enough, because for `Qt5` specifically the use of `ninja` is indirect: the `EB_Qt` easyblock we use for building `Qt5` is based on `ConfigureMake`...

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.13.1-GCCcore-8.3.0.eb_ 

Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2610.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/de5e26656449b2c35cfa131c244d3b34 for a full test report.

File: easybuild/easyblocks/q/qt.py [hunk 1]
```diff
@@ -151,6 +153,12 @@ def configure_step(self):
         ]
         run_cmd_qa(cmd, qa, no_qa=no_qa, log_all=True, simple=True, maxhits=120)
 
+        # Ninja uses all visible cores by default, which can lead to lack of sufficient memory;
+        # so $NINJAFLAGS is set to control number of parallel processes used by Ninja
+        if LooseVersion(self.version) >= LooseVersion('5'):
+            if get_software_root('Ninja'):
+                env.setvar('NINJAFLAGS', '-j%s' % self.cfg['parallel'])
```
Comment (akesandgren) [inline easybuild/easyblocks/q/qt.py:160]: It's likely a bad idea to overwrite NINJAFLAGS like this. User could have set it for some reason. And it should check if "-j" is already in NINJAFLAGS and not override it if it is...

Comment (surak) [reply easybuild/easyblocks/q/qt.py:160]: Should one add this variable on ninja itself? Or make ninja respect parallel?

**Comment by akesandgren** [comment]
This one could use a bit of revisiting from @boegel

**Comment by boegel** [comment]
I did a bit more digging, and apparently `$NINJAFLAGS` is not a generic thing for `Ninja`, it's very specific to `Qt5`...
Ninja itself only picks up a single environment variable according to https://ninja-build.org/manual.html, and that's only to tweak how it reports status.

File: easybuild/easyblocks/q/qt.py (hunk 1 above)
Comment (boegel) [reply easybuild/easyblocks/q/qt.py:160]: I've just figured out that `$NINJAFLAGS` is very specific to `Qt5`, so in that case I'd argue that ignoring `$NINJAFLAGS` if it's set already actually makes total sense (you don't want a random value for a pretty generically named environment variable like `$NINJAFLAGS` to mess up your Qt5 build).

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.8.0-foss-2017b.eb_ 
 * **SUCCESS** _Qt5-5.9.3-foss-2017b.eb_ 
 * **SUCCESS** _Qt5-5.6.0-foss-2016a.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
node2622.swalot.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/2e4eca92bc73a706382a73aa8ae1eef8 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Qt5-5.13.1-GCCcore-8.3.0.eb_ 
 * **SUCCESS** _Qt5-5.14.1-GCCcore-9.3.0.eb_ 
 * **SUCCESS** _Qt5-5.14.2-GCCcore-10.2.0.eb_ 
 * **SUCCESS** _Qt5-5.15.2-GCCcore-10.3.0.eb_ 
 * **SUCCESS** _Qt5-5.15.2-GCCcore-11.2.0.eb_ 

Build succeeded for 5 out of 5 (5 easyconfigs in total)
node3575.doduo.os - Linux RHEL 8.4, x86_64, AMD EPYC 7552 48-Core Processor (zen2), Python 3.6.8
See https://gist.github.com/d07ce43175b3de250035daf99ea4d547 for a full test report.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**